pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -861,16 +1399,46 @@ func (pl *DynamicResources) bindClaim(ctx context.Context, state *stateData, ind
 					logger.V(5).Info("Claim not stored in assume cache", "err", finalErr)
 				}
 			}
-			pl.draManager.ResourceClaims().RemoveClaimPendingAllocation(claim.UID)
+			for _, claimUID := range claimUIDs {
+				pl.draManager.ResourceClaims().RemoveClaimPendingAllocation(claimUID)
+			}
 		}
 	}()
 
-	logger.V(5).Info("preparing claim status update", "claim", klog.KObj(state.claims[index]), "allocation", klog.Format(allocation))
+	// Create the special claim for extended resource backed by DRA?
+	if isExtendedResourceClaim && claim.Name == specialClaimInMemName {
+		// Replace claim template with instantiated claim for the node.
+		if nodeAllocation, ok := state.nodeAllocations[nodeName]; ok && nodeAllocation.extendedResourceClaim != nil {
+			claim = nodeAllocation.extendedResourceClaim.DeepCopy()
+		} else {
+			return nil, fmt.Errorf("extended resource claim not found for node %s", nodeName)
+		}
+
+		// Clear fields which must or can not be set during creation.
+		claim.Status.Allocation = nil
+		claim.Name = ""
+		claim.UID = ""
+		var err error
+		claim, err = pl.clientset.ResourceV1beta1().ResourceClaims(claim.Namespace).Create(ctx, claim, metav1.CreateOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("create claim for extended resources %s: %w", klog.KObj(claim), err)
+		}
+		logger.V(5).Info("created claim for extended resources", "pod", klog.KObj(pod), "node", klog.ObjectRef{Name: nodeName}, "resourceclaim", klog.Format(claim))
+
+		// Track the actual extended ResourceClaim from now.
+		// Relevant if we need to delete again in Unreserve.
+		if err := state.claims.setExtendedResourceClaim(claim); err != nil {
+			return nil, fmt.Errorf("extended resource claim cannot be set for node %s", nodeName)
```

Comment:
```suggestion
			return nil, fmt.Errorf("internal error: update extended ResourceClaim: %w", err)
```

There's no need to include the node name in the error, the node has nothing to do with it.

Perhaps rename `setExtendedResourceClaim` to `updateExtendedResourceClaim` because that explains the pre-requisite (must exist already)?